aws/aws.go

```diff
@@ -8,37 +8,87 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	envVarGuestAccessKeyID     = "GUEST_AWS_ACCESS_KEY_ID"
+	envVarGuestSecretAccessKey = "GUEST_AWS_SECRET_ACCESS_KEY"
+	envVarGuestSessionToken    = "GUEST_AWS_SESSION_TOKEN"
+	envVarHostAccessKeyID      = "HOST_AWS_ACCESS_KEY_ID"
+	envVarHostSecretAccessKey  = "HOST_AWS_SECRET_ACCESS_KEY"
+	envVarHostSessionToken     = "HOST_AWS_SESSION_TOKEN"
+	envVarRegion               = "AWS_REGION"
+)
+
+var (
+	guestAccessKeyID     string
+	guestSecretAccessKey string
+	guestSessionToken    string
+	hostAccessKeyID      string
+	hostSecretAccessKey  string
+	hostSessionToken     string
+	region               string
 )
 
 type Client struct {
 	EC2            *ec2.EC2
 	CloudFormation *cloudformation.CloudFormation
 }
 
-func NewClient() *Client {
+func NewClient() (*Client, error) {
 	a := &Client{}
 
 	{
+		region = os.Getenv(envVarRegion)
```

Comment:
Cool! <3 I was thinking of having a lower level check on e2e-harness were `project.yaml` is parsed, but at this level is great.

Reply:
Thanks! It would be useful to have checks in e2e-harness too. But I think its useful to check here too.